protocol/messenger.go

```diff
@@ -139,9 +139,12 @@ func (interceptor EnvelopeEventsInterceptor) EnvelopeSent(identifiers [][]byte)
 		err := interceptor.Messenger.processSentMessages(ids)
 		if err != nil {
 			interceptor.Messenger.logger.Info("Messenger failed to process sent messages", zap.Error(err))
+		} else {
+			interceptor.EnvelopeEventsHandler.EnvelopeSent(identifiers)
 		}
+	} else {
```

Comment:
This I am not sure I understand, would you mind leaving a comment to explain why it's an `if or else`?

Reply:
Soooo.... right now `processSentMessages` can return error when connection state is offline which means we don't want to execute `interceptor.EnvelopeEventsHandler.EnvelopeSent` but if there is no `else` it will be called anyway. I can move connection state check to `EnvelopeSent` but having a separate `if` or `else` is probably unavoidable in this case.

Reply:
I think it's fine, maybe if you could leave the explanation in a comment in the code that's enough, thank you

Reply:
ok

Reply:
@cammellos done